libp2p/nameresolving/dnsresolver.nim

```diff
@@ -110,7 +114,10 @@ method resolveIp*(
       try:
         let resp = await fut
         for answer in resp.answers:
-          resolvedAddresses.incl(answer.toString())
+          resolvedAddresses.incl(
+            try: answer.toString()
+            except Exception as exc: raise newException(CatchableError, exc.msg)
+          )
       except CancelledError as e:
         raise e
       except ValueError as e:
```

Comment:
We shouldn't handle `ValueError` as it's a `Defect` type.

Reply:
`ValueError* = object of CatchableError`

Pretty sure I caught all the Defect catching attempt as part of Various Cleanups part 1

Reply:
I think it's slated to be a defect, I remember seeing this somewhere - @arnetheduck might know as well.

Reply:
It's a catchable, and that won't change.